rpcs3/Emu/System.cpp

```diff
@@ -318,49 +321,54 @@ void Emulator::Load()
 		LOG_NOTICE(LOADER, "Title: %s", GetTitle());
 		LOG_NOTICE(LOADER, "Serial: %s", GetTitleID());
 
-		// Initialize data/cache directory
-		m_cache_path = fs::get_data_dir(m_title_id, m_path);
-		LOG_NOTICE(LOADER, "Cache: %s", GetCachePath());
+		std::string bdvd_dir;
 
-		// Load custom config-0
-		if (fs::file cfg_file{m_cache_path + "/config.yml"})
+		if (!add_only)
```

Comment:
If I return directly before run() it freezes rpcs3 for a while until we have added all the games.
if I return anywhere before that I get above error